include/swift/AST/Decl.h

```diff
@@ -3175,13 +3175,33 @@ class TypeDecl : public ValueDecl {
 
   void setInherited(ArrayRef<InheritedEntry> i) { Inherited = i; }
 
-  /// Is it possible for this type to lack a Copyable constraint?
+  /// Indicates how "strongly" a TypeDecl will conform to an invertible
+  /// protocol. Supports inequality comparisons and casts to bool.
+  enum CanBeInvertibleResult: unsigned {
```

Comment:
Use an enum class so that the case names are namespaced properly

Reply:
I wrapped the enum in a class for the namespacing, while preserving the behavior that the result of `canBeCopyable` can be cast to bool implicitly. Might be a little gross but it does make the code pretty. Since the conditional case isn't yet being used as I thought, perhaps it should be renamed (again) to `alwaysCopyable` since the conditionally copyable case is always lumped into the not-copyable situations, especially in type lowering.